server/services/unusedDeductionsService.ts

```diff
@@ -18,87 +20,106 @@ export default class UnusedDeductionsService {
   }
 
   /* Wait until calclulated unused deductions matches with adjustments database. */
-  async waitUntilUnusedRemandCreated(nomsId: string, token: string): Promise<boolean> {
+  async waitUntilUnusedRemandCreated(nomsId: string, token: string): Promise<UnusedDeductionMessageType> {
     try {
       let adjustments = await this.adjustmentsService.findByPersonOutsideSentenceEnvelope(nomsId, token)
 
-      const deductions = adjustments.filter(it => it.adjustmentType === 'REMAND' || it.adjustmentType === 'TAGGED_BAIL')
-      if (!deductions.length) {
-        // If there are no deductions then unused deductions doesn't need to be calculated
-        return true
-      }
-      if (this.anyDeductionFromNomis(deductions)) {
-        // won't calculate unused deductions if adjusments are not from DPS.
-        return false
-      }
-
       const unusedDeductionsResponse = await this.calculateReleaseDatesService.calculateUnusedDeductions(
         nomsId,
         adjustments,
         token,
       )
 
       if (unusedDeductionsResponse.validationMessages?.length) {
-        return false
+        if (
+          unusedDeductionsResponse.validationMessages.find(
+            it => it.type === 'UNSUPPORTED_CALCULATION' || it.type === 'UNSUPPORTED_SENTENCE',
+          )
+        ) {
+          return 'UNSUPPORTED'
+        }
+
+        return 'VALIDATION'
       }
       const calculatedUnusedDeducions = unusedDeductionsResponse.unusedDeductions
 
+      const deductions = adjustments.filter(it => it.adjustmentType === 'REMAND' || it.adjustmentType === 'TAGGED_BAIL')
+      if (!deductions.length) {
```

Comment:
This bit didn't need to move. Just the `anyDeductionFromNomis` condition. Otherwise looks good.